tcga_utils/wsi_utils.py

```diff
@@ -499,16 +499,38 @@ def __init__(self, uuid, mpp=0.5):
         tiles = wsi_utils.slide_to_tiles(slide, new_mpp=mpp)
         filtered_tiles = wsi_utils.filter_tiles(tiles)
         download_utils.remove_file(svs_filename)
-        
+
         self.uuid = uuid
         self.tiles = filtered_tiles
         self.mpp = mpp
         self.annotation = {}
         return None
 
+    def tcga_dictionary():
+        """
+        Dictionary for classes/clusters and their corresponding uuid_cases (extracted from table produced by Peter Nguyen).
+        Function is stored within TCGADataset class. It can be callable using tcga_class['class #']. Also note that class number
+        corresponds to the cluster number
+        """
+        tcga_df = pd.read_csv("../tcga_utils/tcga_uuid.csv", usecols = ["uuid_case", "cluster"], index_col = "cluster")
+        tcga_keys = {'class 1' : tcga_df.loc[1],
+                    'class 2' : tcga_df.loc[2],
+                    'class 3' : tcga_df.loc[3],
+                    'class 4' : tcga_df.loc[4],
+                    'class 5' : tcga_df.loc[5],
+                    'class 6' : tcga_df.loc[6],
+                    'class 7' : tcga_df.loc[7],
+                    'class 8' : tcga_df.loc[8],
+                    'class 9' : tcga_df.loc[9],
+                    'class 10' : tcga_df.loc[10],
+                    'class 11' : tcga_df.loc[11],
+                    'class 12' : tcga_df.loc[12]}
+        return tcga_keys
+tcga_class = TCGADataset.tcga_dictionary()
```

Comment:
In my mind, I thought a dataset would represent a single data item. With this in mind, for that datasets uuid, you should go into your dataframe and find out which class the dataset's uuid is in.